src/Quartz/Impl/JobType.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+
+using Quartz.Util;
+
+namespace Quartz.Impl;
+
+/// <summary>
+/// Store the Job Type and FullName for serialization
+/// </summary>
+[Serializable]
+public sealed class JobType
+{
+    private Lazy<Type> type = new(() => throw new InvalidOperationException("Type not defined"));
```

Comment:
Do we want to pay this extra cost for creating a job type:

|        Method |        Mean |     Error |    StdDev |      Median |  Gen 0 | Allocated |
|-------------- |------------:|----------:|----------:|------------:|-------:|----------:|
| Type |  16.9039 ns | 0.1538 ns | 0.1364 ns |  16.8752 ns | 0.0057 |      24 B |
| JobType | 162.0058 ns | 2.1970 ns | 1.9476 ns | 161.9113 ns | 0.1185 |     497 B |

Once initialized the extra cost is negligable:

|        Method |        Mean |     Error |    StdDev |      Median |  Gen 0 | Allocated |
|-------------- |------------:|----------:|----------:|------------:|-------:|----------:|
|      Type |   0.0031 ns | 0.0120 ns | 0.0106 ns |   0.0000 ns |      - |         - |
|      JobType |   4.1255 ns | 0.1113 ns | 0.2646 ns |   4.0576 ns |      - |         - |

Use of **Lazy\<T>** should also have impact on concurrency. Haven't had time to benchmark this.